src/components/Accordion.tsx

```diff
@@ -0,0 +1,142 @@
+import { Css, Icon, useTestIds, Palette } from "src/index";
+import { Dispatch, ReactNode, SetStateAction, useEffect, useState } from "react";
+import { useFocusRing } from "react-aria";
+import { useId } from "@react-aria/utils";
+
+export interface AccordionProps {
+  label?: string;
+  title: ReactNode;
+  children: ReactNode;
+  disabled?: boolean;
+  defaultExpanded?: boolean;
+  size?: AccordionSize;
+  /** Adds a top border (enabled by default) */
+  topSection?: boolean;
+  /** Adds a bottom border (disabled by default) */
+  bottomSection?: boolean;
+  /** Used by AccordionList */
+  allowMultipleExpanded?: boolean;
+  index?: number;
+  currentSelectedIndex?: number;
+  setCurrentSelectedIndex?: Dispatch<SetStateAction<number | undefined>>;
+}
+
+type AccordionSize = "xs" | "s" | "m" | "l";
+
+const accordionSizes: Record<AccordionSize, number> = {
+  "xs": 240,
+  "s": 360,
+  "m": 480,
+  "l": 600
+}
+
+export function Accordion(props: AccordionProps) {
+  const {
+    title,
+    children,
+    size,
+    disabled = false,
+    defaultExpanded = false,
+    topSection = true,
+    bottomSection = false,
+    label = "accordion",
+    allowMultipleExpanded = true,
+    index,
+    setCurrentSelectedIndex,
+    currentSelectedIndex
+  } = props;
+  const testIds = useTestIds({}, label);
+  const id = useId();
+  const [expanded, setExpanded] = useState(defaultExpanded);
+  const { isFocusVisible, focusProps } = useFocusRing();
+
+  useEffect(() => {
+    // Handle only one expanded accordion when selected in AccordionList
+    if (!allowMultipleExpanded && index !== currentSelectedIndex) {
+      setExpanded(false)
+    }
+  }, [allowMultipleExpanded, currentSelectedIndex, index]);
+
+  return (
+    <div
+      {...testIds}
+      css={
+       {...Css.mbPx(4)
```

Comment:
I had to add this margin because on the `AccordionList`, when one accordion was highlighed, it created an overlap with the next one

<img width="1302" alt="image" src="https://user-images.githubusercontent.com/24662091/190401922-8e4cea2f-606c-4e88-a292-4e323f882db5.png">

Reply:
**suggestion:** I would suggest we use an `inset` box-shadow for the focus state. That'd remove the issue of other accordions overlapping the focus styles.